### PR TITLE
Add baryon- and gas-fractions in haloes

### PR DESCRIPTION
**Pull request by MatthieuSchaller**

### CODE REVIEW

**Comment by MatthieuSchaller** [comment]
This needs to be divided by Omega_b / Omega_m on the y-axis to be more useful and hence needs to wait for the cosmology parameters to be read in from the VR catalogues.

**Comment by MatthieuSchaller** [comment]
This is now good to go.

**Comment by MatthieuSchaller** [comment]
Scrap that, we can do much better.

**Comment by MatthieuSchaller** [comment]
Requires https://github.com/SWIFTSIM/velociraptor-python/issues/22 first.

**Comment by JBorrow** [comment]
This should now be possible.

**Comment by MatthieuSchaller** [comment]
All good now. Just need the change to the VR library upstream to fix the look of the x-axis.

gas_fraction_500

**Comment by JBorrow** [comment]
*cries in forgotten $ symbols*

File: eagle-xl/registration.py [hunk 1]
```diff
@@ -164,3 +166,23 @@
         ),
     )
 
+# Now baryon fractions
+
+Omega_m = catalogue.units.cosmology.Om0
+Omega_b = catalogue.units.cosmology.Ob0
+
+M_500 = catalogue.spherical_overdensities.mass_500_rhocrit
```
Comment (JBorrow) [inline eagle-xl/registration.py:174]: This needs to be protected in a similar way as the HI/H2 above.

Comment (MatthieuSchaller) [reply eagle-xl/registration.py:174]: Ok, but we don't have a reasonable replacement here. Do you want me to just propose an array of zeros?

Comment (JBorrow) [reply eagle-xl/registration.py:174]: I think an array of ones here would be best as that wouldn't lead to crashes. Let's do that for now and I'll revisit this once I have actually got the 'only show this plot if you could make it' working properly.

Comment (MatthieuSchaller) [reply eagle-xl/registration.py]: Done. :)

**Comment by MatthieuSchaller** [comment]
BTW, this needed a small tweak to the data https://github.com/SWIFTSIM/velociraptor-comparison-data/pull/42

**Pull request merged**